src/prefect/infrastructure/provisioners/container_instance.py

```diff
@@ -0,0 +1,765 @@
+        table = Panel(
+            dedent(
+                f"""\
+                    Provisioning infrastructure for your work pool [blue]{work_pool_name}[/] will require:
+
+                        Updates in subscription [blue]{self._subscription_name}[/]
+
+                            - Create a resource group in location [blue]{self._location}[/]
+                            - Create an app registration in Azure AD [blue]{self.APP_REGISTRATION_NAME}[/]
+                            - Create/use a service principal for app registration
+                            - Generate a secret for app registration
+                            - Assign Contributor role to service account
+                            - Create Azure Container Instance 'aci-push-pool-container' in resource group [blue]{self.RESOURCE_GROUP_NAME}[/]
+
+                        Updates in Prefect workspace
+
+                            - Create Azure Container Instance credentials block [blue]aci-push-pool-credentials[/]
+                    """
+            ),
+            expand=False,
+        )
+        self._console.print(table)
+        if self._console.is_interactive:
+            if not Confirm.ask(
+                "Proceed with infrastructure provisioning?", console=self._console
+            ):
+                return base_job_template
+
+        with Progress(console=self._console) as progress:
+            task = progress.add_task("Provisioning infrastructure...", total=7)
+            progress.console.print("Creating resource group")
+            await self._create_resource_group()
+            progress.advance(task)
+
+            progress.console.print("Creating app registration")
+            client_id = await self._create_app_registration()
+            progress.advance(task)
+
+            credentials_block_exists = await self._aci_credentials_block_exists(
+                block_name=f"{work_pool_name}-push-pool-credentials", client=client
+            )
+
+            if not credentials_block_exists:
+                progress.console.print("Generating secret for app registration")
+                tenant_id, client_secret = await self._generate_secret_for_app(
+                    app_id=client_id,
+                )
+                progress.advance(task)
+
+                progress.console.print("Creating ACI credentials block")
+                block_doc_id = await self._create_aci_credentials_block(
+                    work_pool_name, client_id, tenant_id, client_secret, client
+                )
+                progress.advance(task)
+            else:
+                progress.console.print("ACI credentials block already exists")
+                block_doc = await client.read_block_document_by_name(
+                    name=f"{work_pool_name}-push-pool-credentials",
+                    block_type_slug="azure-container-instance-credentials",
+                )
+                block_doc_id = block_doc.id
+                progress.advance(task)
```

Comment:
We'll want to advance by 2 here to ensure the amount advanced is asymmetric between the two branches.
```suggestion
                progress.advance(task, advance=2)
```

Reply:
Oops, I forgot to push this:
https://github.com/PrefectHQ/prefect/pull/11275/files#diff-cfa849f982ca38f020a3d71caefe80bd5b7e78dbb3a2186875853964e54e75c3R711-R718

do you prefer using `advance` over that?

Reply:
Yup that will work!